src/tqec/plaquette/library/spatial.py

```diff
@@ -0,0 +1,198 @@
+        match self._plaquette_kind:
+            case "UP":
+                self._moments = self._build_memory_moments_up()
+            case "DOWN":
+                self._moments = self._build_memory_moments_down()
+
+        self._qubit_map = QubitMap(
+            {0: self._qubits.syndrome_qubits[0]}
+            | {i + 1: q for i, q in enumerate(self._qubits.data_qubits)}
+        )
+
+        self._data_init: Basis | None = None
+        self._data_meas: Basis | None = None
+
+    def _get_data_qubits_for_init_or_meas(self) -> list[int]:
+        match self._plaquette_kind:
+            case "UP":
+                return [2] if self._is_corner_trimmed else [1, 2]
+            case "DOWN":
+                return [3] if self._is_corner_trimmed else [3, 4]
+
+    def _get_plaquette_name(self) -> str:
+        parts = [
+            _SpatialCubeArmPlaquetteBuilder.BASE_NAME,
+            self._basis.name,
+            self._plaquette_kind,
+        ]
+        if self._is_reverse:
+            parts.append("REVERSE")
+        if self._is_corner_trimmed:
+            parts.append("CORNER_TRIMMED")
+        if self._data_init is not None:
+            parts.append(f"datainit({self._data_init.name})")
+        if self._data_meas is not None:
+            parts.append(f"datameas({self._data_meas.name})")
+        return "_".join(parts)
+
+    def build(self) -> Plaquette:
+        return Plaquette(
+            self._get_plaquette_name(),
+            self._qubits,
+            ScheduledCircuit(self._moments, schedule=0, qubit_map=self._qubit_map),
+            mergeable_instructions=self.MERGEABLE_INSTRUCTIONS,
+        )
+
+    def _append_ctrl_op_to_data_qubit(self, circuit: stim.Circuit, target: int) -> None:
+        if self._trimmed_qubit is not None and target == self._trimmed_qubit:
+            circuit.append("TICK")
+        else:
+            circuit.append(f"C{self._basis.name}", [0, target], [])
+
+    def _build_memory_moments_up(self) -> list[Moment]:
+        circuit = stim.Circuit()
+        circuit.append("RX", [0], [])
+        circuit.append("RZ", [3], [])
+        circuit.append("TICK")
+        circuit.append("CX", [0, 3], [])
+        circuit.append("TICK")
+        targ_order = [2, 1] if self._is_reverse else [1, 2]
+        self._append_ctrl_op_to_data_qubit(circuit, targ_order[0])
+        circuit.append("TICK")
+        circuit.append("TICK")
+        self._append_ctrl_op_to_data_qubit(circuit, targ_order[1])
+        circuit.append("TICK")
+        circuit.append("CX", [3, 0], [])
+        return list(iter_stim_circuit_without_repeat_by_moments(circuit))
+
+    def _build_memory_moments_down(self) -> list[Moment]:
+        circuit = stim.Circuit()
+        circuit.append("RZ", [1], [])
+        circuit.append("TICK")
+        circuit.append("RZ", [0], [])
+        circuit.append("TICK")
+        circuit.append("CX", [1, 0], [])
+        circuit.append("TICK")
+        targ_order = [4, 3] if self._is_reverse else [3, 4]
```

Comment:
I do not understand how this qubit indexing can work. When trying to do the circuit by myself to check your implementation, I came up with `[3, 2] if self._is_reverse else [2, 3]`. Could you explain your indexing here?

Reply:
Yes! I simply skipped (2) in my head, and that was my mistake. Your indexing is the correct one, thanks for the explanation.